src/arch/riscv/isa/decoder.isa

```diff
@@ -2734,23 +2911,148 @@ decode QUADRANT default Unknown::unknown() {
                         Rd = f16_classify(f16(freg(Fs1_bits)));
                     }}, FloatMiscOp);
                 }
-                0x78: fmv_w_x({{
-                    freg_t fd;
-                    fd = freg(f32(Rs1_uw));
-                    Fd_bits = fd.v;
-                }}, FloatCvtOp);
+                0x78: decode ROUND_MODE {
+                    0x0: decode CONV_SGN {
+                        0x00: fmv_w_x({{
+                            freg_t fd;
+                            fd = freg(f32(Rs1_uw));
+                            Fd_bits = fd.v;
+                        }}, FloatCvtOp);
+                        0x01: fli_s({{
+                            const uint32_t imm[32] = {
```

Comment:
I think the `imm` should be set to a static variable to avoid allocating the array for every instruction execution call.